From: Mitsuya Shibata <mty.shibata@gmail.com>
Date: Sun, 10 Apr 2022 17:10:01 +0900
Subject: src/services: Fix bashisms in
 org.freedesktop.IBus.session.GNOME.service

The "[[ ... ]]" is bash's compound command not in POSIX sh. This style
is problematic in Xorg session in environments where sh is not bash,
suach as Debian and Ubuntu.

Origin: https://github.com/ibus/ibus/pull/2398
Bug: https://github.com/ibus/ibus/issues/2397
Bug-Debian: https://bugs.debian.org/1009256
Bug-Ubuntu: https://launchpad.net/bugs/1968459
---
 bus/services/org.freedesktop.IBus.session.GNOME.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bus/services/org.freedesktop.IBus.session.GNOME.service.in b/bus/services/org.freedesktop.IBus.session.GNOME.service.in
index a99370fad..7f718e3c9 100644
--- a/bus/services/org.freedesktop.IBus.session.GNOME.service.in
+++ b/bus/services/org.freedesktop.IBus.session.GNOME.service.in
@@ -18,7 +18,7 @@ Conflicts=gnome-session@gnome-login.target
 [Service]
 Type=dbus
 # Only pull --xim in X11 session, it is done via Xwayland-session.d on Wayland
-ExecStart=sh -c '@bindir@/ibus-daemon --panel disable $([[ $XDG_SESSION_TYPE == "x11" ]] && echo "--xim")'
+ExecStart=sh -c '@bindir@/ibus-daemon --panel disable $([ "$XDG_SESSION_TYPE" = "x11" ] && echo "--xim")'
 Restart=on-abnormal
 BusName=org.freedesktop.IBus
 TimeoutStopSec=5
